WordPress/gutenberg
# Patterns: Disable the preview option button when editing

Fixes: #53862
Follow-up on: #52427

## What?

This PR ~~removes~~ disables the header device preview button in the pattern edit mode.

![patterns](https://github.com/WordPress/gutenberg/assets/54422211/62f84608-2c18-4262-8f8a-a73ce8f7732c)

## Why?

The pattern edit page should not need this button, as the canvas is resizable by handles, as are the template parts and navigation.

## How?

I have discovered that there is a `FOCUSABLE_ENTITIES` constant that defines whether the current entity is focusable or not.
I moved `constant.js` to the proper location to reuse this constant. This will prevent regression if editable entities are updated in the future.


## Testing Instructions

- Create an unsynced pattern.
- Enter edit mode.
- Make sure that the preview button is ~~not displayed~~ disabled in the header.
